controllers/managedfusionoffering_controller.go

```diff
@@ -320,12 +320,23 @@ func pluginIsReadyToBeRemoved(reconciler *ManagedFusionOfferingReconciler, offer
 
 // This function is a placeholder for offering plugin integration
 func pluginReconcile(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
-	return dfReconcile(reconciler, offering)
+	switch offering.Spec.Kind {
+	case v1alpha1.KindDataFoundation:
+		if _, err := dfReconcile(reconciler, offering); err != nil {
+			return ctrl.Result{}, err
+		}
+	case v1alpha1.KindDataFoundationClient:
+		if _, err := dfcReconcile(reconciler, offering); err != nil {
+			return ctrl.Result{}, err
```

Comment:
`dfcReconcile` is already returning ctrl.Result, why not use it? if its not required dont return it

Reply:
@Madhu-1 is right, the way that it is written wright now is actually a bug

Reply:
updated